### PR TITLE
fix cross_section for paths

### PR DESCRIPTION
**Pull request by joamatab**
fixes https://github.com/gdsfactory/gdsfactory/issues/2226

@95-martin-orion

### CODE REVIEW

**Comment by 95-martin-orion** [comment]
Is this meant to link to #2226? I don't see how it relates to #2227.

File: gdsfactory/port.py [hunk 1]
```diff
@@ -1046,7 +1046,24 @@ def auto_rename_ports_layer_orientation(
 if __name__ == "__main__":
     import gdsfactory as gf
 
-    c = gf.c.straight()
-    p2 = c["o2"]
-    p2.x = 20
+    my_cross_section = gf.CrossSection(sections=[gf.Section(layer=(1, 1), width=2)])
```
Comment (95-martin-orion) [inline gdsfactory/port.py:1049]: Should this be added as a test so that CI notices if it breaks?

Comment (joamatab) [reply gdsfactory/port.py:1049]: yes, we can do that

**Comment by joamatab** [comment]
yes, good catch Orion

I meant :)
https://github.com/gdsfactory/gdsfactory/issues/2226

**Comment by joamatab** [comment]
I also added  a test

**Pull request merged**